Bottom: 280c60267ad5a68f4cc06fd1990d852594424baa
Top:    d7aeaa65262933b961aa57e0760d569c5bcbcf48
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 16:02:05 +0000

Add a new flag, "connected" / CONNECTEDFLAG.

The intended use of this flag is to mark physical connections.


---

diff --git a/src/const.h b/src/const.h
index 66a583f..7cbf194 100644
--- a/src/const.h
+++ b/src/const.h
@@ -178,6 +178,8 @@ pins, indicates that the pin is closer to a horizontal edge and thus
 pinout text should be vertical.
 @item 0x8000 marker
 Marker used internally to avoid revisiting an object.
+@item 0x10000 connected
+If set, this object has been as physically connected by @code{FindConnection()}.
 @end table
 %end-doc */
 
@@ -214,9 +216,10 @@ Marker used internally to avoid revisiting an object.
 #define EDGE2FLAG               0x4000  /* Padr.Point2 is closer to outside edge */
                                         /* also pinout text for pins is vertical */
 #define VISITFLAG               0x8000  /* marker to avoid re-visiting an object */
+#define CONNECTEDFLAG          0x10000  /* flag like FOUND flag, but used to identify physically connected objects (not rats) */
 
 
-#define NOCOPY_FLAGS (FOUNDFLAG)
+#define NOCOPY_FLAGS (FOUNDFLAG | CONNECTEDFLAG)
 
 /* ---------------------------------------------------------------------------
  * PCB flags
diff --git a/src/strflags.c b/src/strflags.c
index 3699c53..f29da5b 100644
--- a/src/strflags.c
+++ b/src/strflags.c
@@ -108,7 +108,8 @@ static FlagBitsType object_flagbits[] = {
   { LOCKFLAG, N ("lock"), ALL_TYPES },
   { EDGE2FLAG, N ("edge2"), ALL_TYPES },
   { FULLPOLYFLAG, N ("fullpoly"), POLYGON_TYPE},
-  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE }
+  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE },
+  { CONNECTEDFLAG, N ("connected"), ALL_TYPES }
 };
 
 static FlagBitsType pcb_flagbits[] = {
